base/testfiles/github-1183.lvt

```diff
@@ -0,0 +1,28 @@
+\input{test2e}
+
+\begin{filecontents}[overwrite]{github-1183.cls}
+\NeedsTeXFormat{LaTeX2e}
+\ProvidesClass{github-1183}
+\LoadClassWithOptions{article}
+
+\DeclareKeys[testwork]{%
+  test .code  = \newcommand{\foo}{#1},
+  test .usage = load,
+  % the following three lines give an equivalent definition of your bertha
+  % option as defined with `\DeclareOption{bertha}{}`
+  bertha .code = {},
+  bertha .value_forbidden:n = true,
+  bertha .usage = load,
+  % if you use this unknown handler, you get "test" and "bertha" as unknown
+  % options, if you omit it you get "a4paper" as unknown option
+  unknown .code = {}
+}
+\ProcessKeyOptions[testwork]
+
+\endinput
+\end{filecontents}
+\documentclass[a4paper,test=wtf,bertha]{github-1183}
```

Comment:
Use an extra option which is unknown to both `article` and `githu-1183` classes, to trigger the "Unused global option(s)" waning?

Reply:
From a conceptual point I think that global options should be handled as follows:
- if a package uses a global option it is marked as used (but still offered to other packages as a global option)
- if no package uses a global option there should be an "unused global option warning"

The big question is what does this mean if a a package defines an unknown option handler? 
- Given that for the default unknown option handler (raising an unknown option error) global options are ignored one can argue that the same should happen if a different unknown option handler is implemented 
- Another possibility is to pass global options to an unknown option handler (if it is not the default) and consider them used if that happens (that means that with one such package we would automatically use all global options and there is no warning ever)
- Finally one could implement some sort of a mix of the two by offering an interface to be used in the unknown option handler to explicitly mark a global option as used if it is passed through and is handled (not sure how that should look like exactly)

Reply:
> We still shouldn't see a warning?

@josephwright I'm not sure which behavior is expected (#1183 has some discussions on it), but with current proposed change, if `aaapaper` is used as class option, it will remain in unused options list. See https://github.com/muzimuzhi/latex2e/commit/587588348942bd38a77fdcf20eb4e836cda38bed.